src/Tables.jl

```diff
@@ -241,10 +241,18 @@ to provide useful default behaviors (allows any `AbstractColumns` to be used lik
 """
 struct Columns{T} <: AbstractColumns
     x::T
+
+    function Columns{T}(x::T) where {T}
+        istable(x) ? new{T}(x) : throw(ArgumentError("Columns can only wrap an object for which `Tables.istable` is true"))
```

Comment:
Hmmm, I don't think I like this restriction. In general, we discourage use of `Tables.istable` and encourage instead to just call `Tables.rows` or `Tables.columns` as needed and code will error if the input isn't valid.

We probably need more docs/usage examples around how to make sure you do `Tables.Column(Tables.columns(x))` to ensure everything works properly. @nalimilan and I recently had a lengthy discussion of adding some additional convenience methods around all this, I'll have to see if I can find it and put the results of our discussion up in an issue.

Reply:
Like I said, the current behavior can lead to some pretty weird stuff.  Regardless, I'd be happy to take this out if we can get it merged, it could always be changed later.